File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/database/DatabaseSwitchRequest.java
package com.mieai.qqbot.admin.database;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

public record DatabaseSwitchRequest(
        @Positive long expectedRevision,
        @Valid @NotNull DatabaseCandidateRequest candidate) {
}

File: qqbot-module-database-support/src/main/kotlin/com/mieai/qqbot/admin/database/DatabaseSwitchRequest.kt
package com.mieai.qqbot.admin.database

import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive

data class DatabaseSwitchRequest(
    @field:Positive val expectedRevision: Long,
    @field:Valid @field:NotNull val candidate: DatabaseCandidateRequest?,
)
